### PR TITLE
Remove json validation and most null value handling

### PR DESCRIPTION
**Pull request by drwill-ms**
Other lessons learned working on STJ:
- Why validate models when the service will validate them?
- We actually do want to include nulls and empty lists in the case where a user wants to unset them.

### CODE REVIEW

File: e2e/test/iothub/service/ConfigurationsClientE2ETests.cs [hunk 1]
```diff
@@ -39,11 +40,11 @@ public async Task ConfigurationOperations_Work()
                     Priority = 2,
                     Labels = { { "labelName", "labelValue" } },
                     TargetCondition = "deviceId='fakeDevice'",
-                    Content = new ConfigurationContent
+                    Content =
```
Comment (drwill-ms) [inline e2e/test/iothub/service/ConfigurationsClientE2ETests.cs:43]: Has default instance.

File: e2e/test/iothub/service/ConfigurationsClientE2ETests.cs [hunk 2]
```diff
@@ -217,5 +212,83 @@ await FluentActions
                 }
             }
         }
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task ConfigurationsClient_ClearLabels()
```
Comment (drwill-ms) [inline e2e/test/iothub/service/ConfigurationsClientE2ETests.cs:218]: Add 2 tests to ensure unsetting labels (2 ways) works.

File: iothub/service/src/Configurations/Models/Configuration.cs [hunk 3]
```diff
@@ -35,26 +33,26 @@ public Configuration(string configurationId)
         /// <summary>
         /// The unique identifier of the configuration.
         /// </summary>
-        [JsonProperty("id", Required = Required.Always)]
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/Models/Configuration.cs:38]: Let server validate.

File: iothub/service/src/Configurations/Models/Configuration.cs [hunk 4]
```diff
@@ -35,26 +33,26 @@ public Configuration(string configurationId)
...
+        [JsonProperty("id")]
         public string Id { get; internal set; }
 
         /// <summary>
         /// The schema version of the configuration.
         /// </summary>
-        [JsonProperty("schemaVersion", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/Models/Configuration.cs:44]: Will never be null.

File: iothub/service/src/Configurations/Models/Configuration.cs [hunk 5]
```diff
@@ -35,26 +33,26 @@ public Configuration(string configurationId)
...
+        [JsonProperty("schemaVersion")]
         public string SchemaVersion { get; } = "1.0";
 
         /// <summary>
         /// The key-value pairs used to describe the configuration.
         /// </summary>
-        [JsonProperty("labels", NullValueHandling = NullValueHandling.Ignore)]
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/Models/Configuration.cs:50]: We actually want to include null to enable unsetting the values.

File: iothub/service/src/Configurations/Models/Configuration.cs [hunk 6]
```diff
@@ -92,30 +90,21 @@ public Configuration(string configurationId)
...
+        // NewtonsoftJsonETagConverter is used here because otherwise the ETag isn't serialized properly.
+        [JsonConverter(typeof(NewtonsoftJsonETagConverter))]
         public ETag ETag { get; set; }
-
-        /// <summary>
-        /// For use in serialization.
-        /// </summary>
-        /// <seealso href="https://www.newtonsoft.com/json/help/html/ConditionalProperties.htm#ShouldSerialize"/>
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        public bool ShouldSerializeLabels()
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/Models/Configuration.cs:116]: If the user removes the labels or sets to null, we want to send an empty object along.

File: iothub/service/src/Configurations/Models/ImportConfiguration.cs [hunk 7]
```diff
@@ -21,7 +21,7 @@ public ImportConfiguration(string configurationId)
         /// <summary>
         /// The type of registry operation and ETag preferences.
         /// </summary>
-        [JsonProperty("importMode", DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
```
Comment (drwill-ms) [inline iothub/service/src/Configurations/Models/ImportConfiguration.cs:24]: We didn't need this for STJ. Don't think we need it for NSJ, either.

File: iothub/service/src/Http/HttpMessageHelper.cs [hunk 8]
```diff
@@ -93,13 +93,13 @@ internal static void ConditionallyInsertETag(HttpRequestMessage requestMessage,
                 // "Perform this operation only if the entity is unchanged"
                 // Sends the If-Match header with a value of the ETag.
                 string escapedETag = EscapeETag(eTag.ToString());
-                requestMessage.Headers.IfMatch.Add(new EntityTagHeaderValue(escapedETag));
+                requestMessage.Headers.IfMatch.Add(new EntityTagHeaderValue(escapedETag, true));
```
Comment (drwill-ms) [inline iothub/service/src/Http/HttpMessageHelper.cs:96]: second parameter indicates this is a weak etag, of which all IoT hub eTags are.

File: e2e/test/iothub/service/ConfigurationsClientE2ETests.cs [hunk 9]
```diff
@@ -217,5 +212,83 @@ await FluentActions
...
+                }
+            }
+        }
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task ConfigurationsClient_NullOutLabels()
+        {
+            // arrange
+
```
Comment (patilsnr) [inline e2e/test/iothub/service/ConfigurationsClientE2ETests.cs:260]: nit: do we want this extra blank line?

**Review by patilsnr — APPROVED** [review]

**Review by tmahmood-microsoft — APPROVED** [review]
Looks good to me

File: e2e/test/iothub/service/ConfigurationsClientE2ETests.cs (hunk 9 above)
Comment (drwill-ms) [reply e2e/test/iothub/service/ConfigurationsClientE2ETests.cs:260]: I usually include a blank line when the comment refers to a section of statements.

An adjacent comment usually indicates it is relevant only to the next line.

**Review by brycewang-microsoft — APPROVED** [review]

**Review by abhipsaMisra — COMMENTED** [review]
👍🏽

**Pull request merged**